Add ISO8583-connector-configuration.md (#1227)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md

> # ISO8583 Connector Configuration

Reviewer: Rename to "Setting up ISO8583" or something similar.

Author: Done.

ISO8583 is an international standard for financial transaction card originated interchange messaging. It is the International Organization for Standardization standard for systems that exchange electronic transactions initiated by cardholders using payment cards.

> Typically, whenever we use a credit card or debit card or an ATM card the data travels from one system to another system. A card based trantraction typically needs to travel between number of systems.

Reviewer: ```suggestion
Typically, whenever we use a credit card, debit card, or ATM card, the data travels from one system to another system. A card-based transaction typically needs to travel between a number of systems.
```

Author: Done.

> ISO8583 is a financial transaction card originated in touch message message specifications adopted by a wide segment of payment industry which include acquiring and issuers.

Reviewer: ```suggestion
ISO8583 is a financial transaction card originated in touch message specifications adopted by a wide segment of the payment industry, which includes acquiring and issuers.
```

What do you mean by "acquiring" here?

Author: Changed as shown bellow.
"ISO 8593 is Financial Transaction card originated, Interchange message specifications adopted by wide segment of Payment Industry (including Acquirers and Issuers)."

## Setting up the environment

> Before you start configuring the ISO8583 connector you also need WSO2 MI and we refer to that location as <PRODUCT_HOME>.

Reviewer: ```suggestion
Before you start configuring the ISO8583 connector, you also need WSO2 MI, and we refer to that location as <PRODUCT_HOME>.
```

Author: Done.

To configure the ISO8583 connector, copy the following client libraries from the given locations to the `<PRODUCT_HOME>/repository/components/lib` directory.

* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 

> >NOTE : For testing purposes, you need to have a test server (basically a java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer)

Reviewer: ```suggestion
> **Note** : For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer).
```

Author: Updated.

> Now that you have connected to test server, For more information, see

Reviewer: ```suggestion
Now you have connected to the test server. For more information, see
```

Merge the lines below with this one.

Author: Done.

## Setting up the environment

Before you start configuring the ISO8583 connector, you also need WSO2 MI, and we refer to that location as <PRODUCT_HOME>.

To configure the ISO8583 connector, copy the following client libraries from the given locations to the `<PRODUCT_HOME>/repository/components/lib` directory.

* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 

## Configure the test server

For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer). 

> You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.For more information on the ISO8583 standard, see [ISO8583 Documentation](https://en.wikipedia.org/wiki/ISO_8583).

Reviewer: ```suggestion
You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag. For more information on the ISO8583 standard, see [ISO8583 Documentation](https://en.wikipedia.org/wiki/ISO_8583).
```
Wikipedia is not so reliable. Do we have a more reliable link?

Author: Updated.

# Setting up ISO8583 Connector 

> ISO8583 is an international standard for financial transaction card originated interchange messaging. It is the International Organization for Standardization standard for systems that exchange electronic transactions initiated by cardholders using payment cards.

Reviewer: "ISO8583 is an international standard for financial transaction card originated interchange messaging." Can we simplify this? I see some similar text in wikipedia and it's a bit too complex. Can we simplify this definition?

Author: Updated.

Typically, whenever we use a credit card, debit card, or ATM card, the data travels from one system to another system. A card-based transaction typically needs to travel between a number of systems.

> ISO 8593 is Financial Transaction card originated, Interchange message specifications adopted by wide segment of Payment Industry (including Acquirers and Issuers).

Reviewer: The "Financial Transaction card originated," part doesn't make much sense here. Can we reword that? 

Also change to ISO8583.

Author: Updated.

## Setting up the environment

Before you start configuring the ISO8583 connector, you also need WSO2 MI, and we refer to that location as <PRODUCT_HOME>.

To configure the ISO8583 connector, copy the following client libraries from the given locations to the `<PRODUCT_HOME>/repository/components/lib` directory.

* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 

## Configure the test server

For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer). 

> You can include required header information within the header tag in . It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.

Reviewer: ```suggestion
You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.
```

Is there some example we can add here or link that we can point to regarding this? It sounds complicated so its better if we can give some more instruction.

Author: Simply added reference document for this part.(https://en.wikipedia.org/wiki/ISO_8583).

This section (Configure the test server) not mandatory configuration for setting up the connector. If customer need to set up this connector with their scenarios, they should have their own ISO8583 back-ends. This sample Java server program just only simulates the ISO8583 mock back end behaviors.

If you use the [sample Java server program](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer) to send an ISO8583 request with a header value from the connector, you need to update the iso87ascii.xml file with the relevant headerLength information.

> The ISO8583 connector uses the jpos library, which is a third party library that provides a high-performance bridge between card messages generated at point of sale terminals, ATMs, and internal systems across the entire financial messaging network. The jposdef.xml file has the field definitions of standard ISO8583 messages. According to the field definitions, each ISO8583 message in XML format coming from the REST client is packed and sent to the test server. Therefore, you need to create a file called jposdef.xml (with the contents given here) in the <EI_HOME> directory.

Reviewer: The "with the contents given here" part is confusing. What contents exactly?

Author: Added link of the repository.

 
Now you have connected to the test server. For more information, see

> [ISO8583 Connector Example](ISO8583-connector-example.md).

Reviewer: Please merge these two lines.

Author: Done.
